src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java

```diff
@@ -53,6 +53,10 @@
     static {
         CHECK_TO_ANCESTOR_NODES_MAP.put("AvoidNestedBlocks",
                 Arrays.asList(TokenTypes.SLIST));
+        CHECK_TO_ANCESTOR_NODES_MAP.put("EmptyStatement",
+                Arrays.asList(TokenTypes.METHOD_DEF));
```

Comment:
METHOD_DEF is too big scope and doesn't cover statements in constructors, lambdas, static blocks, etc. [Example AST](http://rveach.no-ip.org/checkstyle/checkstyle.php?action=view&config=72fba57f86eaf7df54f418b381c28422&code=93063b278d27defddea640338e439801&checkstyle=checkstyle-8.35-all.jar&printTree=true&printSuppressions=).

Why can't we use parent node and `checkNamesForContextStrategyByTokenOrParent` property?

Reply:
@baratali `checkNamesForContextStrategyByTokenOrParent` is ok, sorry, maybe it's my oversight.